packages/shared/src/components/post/smartPrompts/PromptButtons.tsx

```diff
@@ -73,9 +73,21 @@ export const PromptButtons = ({
   const { data, isLoading } = usePromptsQuery();
   const { flags: settingFlags } = useSettingsContext();
   const { prompt: promptFlags } = settingFlags || {};
+  const lastPrompt = settingFlags?.lastPrompt;
   const prompts = useMemo(() => {
-    return data?.filter((prompt) => promptFlags?.[prompt.id] !== false);
-  }, [data, promptFlags]);
+    const filteredPrompts = data?.filter(
+      (prompt) => promptFlags?.[prompt.id] !== false,
+    );
+    if (lastPrompt) {
+      filteredPrompts?.unshift(
+        filteredPrompts?.splice(
+          filteredPrompts.findIndex((item) => item.id === lastPrompt),
+          1,
+        )[0],
+      );
+    }
```

Comment:
```suggestion
    if (filteredPrompts && lastPrompt) {
      const latPromptIndex = filteredPrompts.findIndex(
        (item) => item.id === lastPrompt,
      );
      const [lastPromptItem] = filteredPrompts.splice(latPromptIndex, 1);

      filteredPrompts.unshift(lastPromptItem);
    }
```

same logic, just think variables help a bit with readability

Reply:
We can probably even introduce a generic function called `moveToFront` that will accept a function that will evaluate the criteria.